### ISSUE TITLE
Enable admins to prevent specific usernames being chosen

### ISSUE THREAD

**Issue by hughrun** [bookwyrm-social/bookwyrm#2619]
**Is your feature request related to a problem? Please describe.**
When finalising the "secure mode Mastodon" fix, it occurred to a couple of us that it would be useful to reserve certain usernames to prevent them from being used.

For example a server admin may want to prevent users from registering misleading usernames like "admin", "safety", "official" etc. This should be an editable list of usernames that can be determined by each server admin since it will be contextual to each user-base and language to a large extent.

**Describe the solution you'd like**
A 'reserved username list' that is checked when a new user attempts to register: if their chosen username is in the reserved list the should be presented with a clear error requiring them to choose another name. This could simply repurpose the functionality when a username is already being used, or potentially we may want a slightly different message since they probably *aren't* being used.

If the administrator wants to use one of the reserved names in future to create an "official account" for example, they could remove the name from the list, restart services and quickly register it.

**Describe alternatives you've considered**
None but I'm open to any!

**Additional context**

**Comment by jaschaurbach** [comment]
I think `bookwyrm.instance.actor` should be hardcoded into the list.

**Comment by rokejulianlockhart** [comment]
Very important. Most moderators probably don't want to have to deal with moderation incidents posted by @AdolfGuessWho@example.com.

**Comment by hughrun** [comment]
hmm I hadn't thought of that @rokejulianlockhart. Now I'm wondering if we should overcomplicate this and have two checks for reserved-listing, something like `USERNAME_MUST_NOT_EQUAL` and `USERNAME_MUST_NOT_CONTAIN`.

**Comment by jaschaurbach** [comment]
@hughrun Charles Dickens would not be happy if it would be `USERNAME_MUST_NOT_CONTAIN`

**Comment by rokejulianlockhart** [comment]
@jaschaurbach, good point. I think that it's a useful feature but would necessitate reservation to a point that can't easily be expected of most moderators. Additionally, supporting a USERNAME_MUST_NOT_EQUAL with regex support would probably provide superior versatility for both scenarios.
